src/test/crimson/test_messenger.cc

```diff
@@ -261,8 +261,11 @@ static seastar::future<> test_echo(unsigned rounds,
   }).then([server2] {
     logger().info("server2 shutdown...");
     return server2->shutdown();
-  }).then([server1, server2, client1, client2] {
+  }).then([] {
     logger().info("test_echo() done!\n");
+  }).handle_exception([server1, server2, client1, client2] (auto eptr) {
+    logger().error("test_echo() failed: got exception {}", eptr);
```

Comment:
none of these captured variables is used. did you plan to print them out as well?

Reply:
This makes sure that servers and clients are not destructed before printing the exceptions. Else the assertion error will terminate the test before being able to print the useful information.